idiss-csharp/IdissLib/Class1.cs

```diff
@@ -0,0 +1,333 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.Json;
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+
+namespace IdissLib
+{
+
+    /// Here we provide classes that match some of the Rust types that involved with identity validation and creation. 
+    /// Most of the field names of these classes are chosen so that they match JSON fields of JSON serialization of the corresponding Rust types.
+    public class YearMonth {
+        public string str{get;set;}
+
+        public YearMonth (string s) {
+            str = s;
+        }
+    }
+
+    public class AttributeTag {
+        public string tag{get;set;}
+
+        public AttributeTag(string t)
+        {
+            tag = t;
+        }
+    }
+
+    public class Attribute {
+        public string attribute{get;set;}
+
+        public Attribute(string a)
+        {
+            attribute = a;
+        }
+    }
+
+    public class AttributeList
+    {
+        public YearMonth validTo{get;set;}
+        public YearMonth createdAt{get;set;}
+        public byte maxAccounts{get;set;}
+        public Dictionary<AttributeTag, Attribute> chosenAttributes{get;set;}
+    }
+
+    public class PublicKey {
+        public string schemeId{get;set;}
+        public string verifyKey{get;set;}
+    }
+
+    public class PublicKeys {
+        public Dictionary<string, PublicKey> keys{get;set;}
+        public byte threshold{get;set;}
+    }
+
+    public class PubInfoForIp {
+        public string idCredPub{get;set;}
+        public string regId{get;set;}
+        public PublicKeys publicKeys{get;set;}
+    }
+
+    public class IpArData {
+        public string encPrfKeyShare{get;set;}
+        public string proofComEncEq{get;set;}
+    }
+
+    public class ChoiceArData {
+        public HashSet<UInt32> arIdentities{get;set;}
+        public byte threshold{get;set;}
+    }
+
+    public class PreIdentityObject {
+        public PubInfoForIp pubInfoForIp{get;set;}
+        public Dictionary<string, IpArData> ipArData{get;set;}
+        public ChoiceArData choiceArData{get;set;}
+        public string idCredSecCommitment{get;set;}
+        public string prfKeyCommitmentWithIP{get;set;}
+        public List<string> prfKeySharingCoeffCommitments{get;set;}
+        public string proofsOfKnowledge{get;set;}
+    }
+
+    public class IdentityObject {
+        public PreIdentityObject preIdentityObject{get;set;}
+        public AttributeList attributeList{get;set;}
+        public string signature{get;set;}
+    }
+
+    public class VersionedPreIdentityObject {
```

Comment:
As you have multiple `Versioned` classes, it should be possible to simplify it using generics. You should be able to do something like:
```
class Versioned<T> 
{
	public UInt32 v { get; set; }
	public T value { get; set; }
}
```
and then use it like: `Versioned<PreIdentityObject>`.